components/nimbus/src/lib.rs

```diff
@@ -841,6 +841,184 @@ mod test_schema_bw_compat {
         assert_eq!(exp.app_id, Some("org.mozilla.fenix".to_string()));
         assert_eq!(exp.channel, Some("nightly".to_string()));
     }
+
+    #[test]
+    /// Migrating v1 to v2 involves finding enrollments and experiments that
+    /// don't contain all the feature_id stuff they should and discarding.
+    fn test_migrate_v1_to_v2() -> Result<()> {
```

Comment:
This a first attempt before I realized I needed to be doing these tests in bulk.  I'm inclined to get rid of it, since it's pretty duplicative, however it is a bit of extra coverage which uses slightly different APIs and further more tests valid and invalid stuff in the same batch.  Thoughts?

Reply:
I agree. 

I've suggested a `create_old_database` method below, to help recreate an old version of the database. Perhaps when you write that you could revisit this, refactoring it in terms of that method.

Reply:
I like the idea of being able to recreate old versions of the schema for testing migrations. At one point I was hoping that enum variants in Rust would be the way to go with this, but just having a test method to create an old version of the database is probably simpler to implement and understand.

Reply:
Refactored this test to use create_old_database.  I'll file a separate ticket where we can discuss how best to manage old database schemas.